proxy.go

```diff
@@ -368,6 +369,7 @@ func realMain() {
 
```

Comment:
Another related query - in the go func of `range results`, we have a fatal log in that func if we get an err - does that DTRT wrt to quitting?

Reply:
`results` is the error channel so currently if the proxy fails to serve (for any reason), that error gets logged here. Whether the proxy should exit on error here is debatable I think since it's possible the error is actually due to the client side or it could be that the proxy has hit some process limit. But in the latter case, it might be able to recover as containers are killed.

wdyt @sboeuf, @bergwolf - should we add an `os.Exit(1)` here?